src/main/java/net/coderbot/iris/gl/program/ProgramBuilder.java

```diff
@@ -35,15 +36,22 @@ public static ProgramBuilder begin(String name, @Nullable String vertexSource, @
 			throw new RuntimeException("Failed to compile vertex shader for program " + name, e);
 		}
 
+		try {
+			geometry = new GlShader(ShaderType.GEOMETRY, name + ".gsh", geometrySource, EMPTY_CONSTANTS);
```

Comment:
This should NOT assume that the geometry shader is always present. Instead, the geometry shader should only be created if it is present, and later should only be attached if it is actually created.